lib/wegift/models/remote_code.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'uri'
+
+module Wegift
+  class RemoteCode < Response
+    attr_reader :url
+
+    # response/success
+    attr_reader :amount, :barcode_format, :barcode_string, :code, :expiry_date,
+      :pin, :type
+
+    def initialize(url:)
+      @url = url
+    end
+
+    def self.get(url)
+      uri = URI(url)
+      conn = Faraday.new(url: "#{uri.scheme}://#{uri.host}")
+
+      response =  conn.get("#{uri.path}?format=json") do |req|
+        req.headers['Content-Type'] = 'application/json'
+      end
+
+      remote_code = self.new(url: url)
+      remote_code.parse(response)
+
+      remote_code
+    end
+
+    def parse(response)
+      super
+
+      if is_successful?
+        @amount = @payload['e_code']['amount']
+        @barcode_format = @payload['e_code']['barcode_format']
+        @barcode_string = @payload['e_code']['barcode_string']
+        @code = @payload['e_code']['code']
+        @expiry_date = @payload['e_code']['expiry_date']
+        @pin = @payload['e_code']['pin']
+        @type = @payload['e_code']['type']
```

Comment:
Why not just `Wegift::RemoteCode.new(@payload['e_code'])`? And why does this response has everything embedded in a `e_code`?

Reply:
this response is basically a voucher redeemed. you can look at the response from their api and it seems they just tried to shoe-horn it to get rid of the old order workflow

Reply:
Still we can simplidy and do:
```
    def parse(response)
      super

      Wegift::RemoteCode.new(@payload['e_code'])
    end
```

Reply:
Don't forget to return it on line 29

Reply:
I don't understand why we need to instantiate a whole new class to set some attributes. Moreover, the new instance of `Wegift::RemoteCode` would have its `is_successful?` method returning false and we are using it to handle failure up the call chain.